pandas/tests/extension/base/reshaping.py

```diff
@@ -173,6 +173,20 @@ def test_merge(self, data, na_value):
                  dtype=data.dtype)})
         self.assert_frame_equal(res, exp[['ext', 'int1', 'key', 'int2']])
 
+        # GH  23020
```

Comment:
can you add as a new test.

this should not be Int64 specific, rather be structured to generically work.

Reply:
The test will need to be updated to merge on the extension array data's column.